### PR TITLE
Issue 1/authentication

### PR DESCRIPTION
**Pull request by samshuster**
Authentication added for the newer version of rest-client.

Some backwards incompatible changes IF people outside of this project were directly using
the DatabricksRestClient... which they shouldn't be.

### CODE REVIEW

**Comment by samshuster** [comment]
This will fix #1 (this is how you can have auto closing!)

Also integration tests running now.

File: src/main/java/com/edmunds/rest/databricks/restclient/DatabricksRestClientImpl425.java [hunk 1]
```diff
@@ -41,18 +43,29 @@
 /**
  * To run on CDH5.7.1, use httpclient4.2.5 version API
```
Comment (javamonkey79) [inline src/main/java/com/edmunds/rest/databricks/restclient/DatabricksRestClientImpl425.java]: Unrelated to this PR, but, maybe we should remove this comment?

Comment (samshuster) [reply src/main/java/com/edmunds/rest/databricks/restclient/DatabricksRestClientImpl425.java]: Sure, I will modify it to just say that this is an older httpclient implementation.

**Comment by samshuster** [comment]
After working on examples for readme, realized that the service factory creation could be cleaned up a bit. Going to add that to this pr

**Comment by samshuster** [comment]
Ok builder added. I think we should be good to go now, pending review of course

File: src/main/java/com/edmunds/rest/databricks/DatabricksServiceFactory.java [hunk 2]
```diff
@@ -144,4 +154,85 @@ public DbfsService getDbfsService() {
...
+      return this;
+    }
+
+    /**
+     * Builds a DatabricksServiceFactory. Conducts basic validation.
+     *
+     * @return the databricks service factory object
+     */
+    public DatabricksServiceFactory build() {
+      if (token != null) {
```
Comment (javamonkey79) [inline src/main/java/com/edmunds/rest/databricks/DatabricksServiceFactory.java]: Do you think null check is sufficient, or, should it also check blank?

File: src/main/java/com/edmunds/rest/databricks/DatabricksServiceFactory.java [hunk 3]
```diff
@@ -64,17 +76,35 @@ public DatabricksServiceFactory(String username, String password, String host, i
...
 
+  /**
+   * Create a databricks service factory using personal token authentication instead.
+   *
+   * @param personalToken your personal token
+   * @param host the databricks host
+   * @param maxRetry the maximum number of retries
+   * @param retryInterval the retry interval between each attempt
+   */
+  @Deprecated
```
Comment (javamonkey79) [inline src/main/java/com/edmunds/rest/databricks/DatabricksServiceFactory.java]: Maybe useful to leave a [deprecated java doc](https://docs.oracle.com/javase/7/docs/technotes/guides/javadoc/deprecation/deprecation.html#javadoc_tag)?

Comment (samshuster) [reply src/main/java/com/edmunds/rest/databricks/DatabricksServiceFactory.java]: thanks for a pointer, will add

File: src/main/java/com/edmunds/rest/databricks/DatabricksServiceFactory.java (hunk 2 above)
Comment (samshuster) [reply src/main/java/com/edmunds/rest/databricks/DatabricksServiceFactory.java]: added a comment that this was not intended to be exhaustive validation, and is instead just determining which client to build.

Also added blank checks.

**Pull request merged**